pkg/provider/branch/trunk/trunk.go

```diff
@@ -425,29 +435,20 @@ func (t *trunkENI) CreateAndAssociateBranchENIs(pod *v1.Pod, securityGroups []st
 	return newENIs, nil
 }
 
-// DeleteAllBranchENIs deletes all the branch ENIs associated with the trunk and all the ENIs present in the cool down
-// queue, this is the last API call to the the Trunk ENI before it is removed from cache
-func (t *trunkENI) DeleteAllBranchENIs() {
-	// Delete all the branch used by the pod on this trunk ENI
+// DisassociateAllBranchENIs removes association between all the branch ENIs with the trunk. The branch ENIs will be deleted by the node termination finalizer
+// This is the last API call to the the Trunk ENI before it is removed from cache
+func (t *trunkENI) DisassociateAllBranchENIs() {
+	// Disassociate all the branch used by the pod on this trunk ENI, it will be cleaned up by the finalizer routine
 	// Since after this call, the trunk will be removed from cache. No need to clean up its branch map
 	for _, podENIs := range t.uidToBranchENIMap {
 		for _, eni := range podENIs {
-			err := t.deleteENI(eni)
+			err := t.ec2ApiHelper.DisassociateTrunkInterface(&eni.AssociationID)
```

Comment:
Since this is called upon node deletion, we can afford some retries. Can you put the API call into retry.OnError()? EC2 APIs are eventually consistent and could fail intermittent for any reason. Having a 3/5 retries when disassociating branch ENI from trunk may better safeguard the consistency on our side.

Reply:
Another option is using retry in the wrapper which calls EC2 but I personally prefer adding retry in the immediate caller.